src/TrackGenerator.cpp

```diff
@@ -268,9 +273,27 @@ double TrackGenerator::getZCoord() {
 
 
 /**
+ * @brief Return the array used to store the FSR volumes
+ * @return _FSR_volumes the FSR volumes array indexed by FSR ID
+ */
+FP_PRECISION* TrackGenerator::getFSRVolumesBuffer() {
+#pragma omp critical
```

Comment:
It's critical because you don't want multiple threads to be allocating a new FSR volumes array. This function should be called infrequently so it shouldn't have any effect on performance.